src/Microsoft.Health.Fhir.Core/Features/Definition/SearchableSearchParameterDefinitionManager.cs

```diff
@@ -148,5 +156,16 @@ private bool UsePartialSearchParams(SearchParameterInfo parameter)
                    _fhirReqeustContextAccessor.RequestContext.IncludePartiallyIndexedSearchParams &&
                    parameter.IsSupported;
         }
+
+        private bool IsEnabled(SearchParameterInfo parameter)
+        {
+            if (parameter.Code == "_type")
+            {
+                return true;
+            }
+
+            var searchParameterStatuses = _statusManager.GetAllSearchParameterStatus(default).ConfigureAwait(false).GetAwaiter().GetResult();
```

Comment:
Will this line cause a trip to the DB everytime we check if a searchparameter IsEnabled?  I just want to understand if there are any negative perf implications.

Reply:
Correct, it would. I forgot to pull that outside of the select statement to reduce it to calling once for all of them. Really it would be better to refactor in this area to combine the status management with the definition manager and just have it use the in memory collection that gets updated periodically already.